usb/ehci/ehci.c

```diff
@@ -485,14 +485,23 @@ static int ehci_irqHandler(unsigned int n, void *data)
 {
 	hcd_t *hcd = (hcd_t *)data;
 	ehci_t *ehci = (ehci_t *)hcd->priv;
+	uint32_t currentStatus;
 
-	ehci->status = *(hcd->base + usbsts);
-	*(hcd->base + usbsts) = ehci->status & 0x1f;
+	do {
+		currentStatus = *(hcd->base + usbsts);
```

Comment:
Maybe move this one before the loop? Right now on second iteration we do `currentStatus = *(hcd->base + usbsts); `two times in the row